### PR TITLE
tests/provider: Update resource testing to 0.12 syntax (V Resources)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Related: #8950, #14417

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing: Handled via daily acceptance testing

### CODE REVIEW

**Comment by YakDriver** [comment]
Good to go!

File: aws/resource_aws_vpc_test.go [hunk 1]
```diff
@@ -573,6 +573,10 @@ func TestAccAWSVpc_classiclinkDnsSupportOptionSet(t *testing.T) {
 const testAccVpcConfig = `
 resource "aws_vpc" "test" {
   cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = "terraform-testacc-vpc"
+  }
```
Comment (bflad) [inline aws/resource_aws_vpc_test.go:579]: This addition is extraneous for a basic test and seems to have caused this new test failure:

```
=== CONT  TestAccAWSVpc_basic
TestAccAWSVpc_basic: resource_aws_vpc_test.go:108: Check 4/13 error: aws_vpc.test: Attribute 'tags.%' expected "0", got "1"
--- FAIL: TestAccAWSVpc_basic (17.45s)
```

```suggestion
```

File: aws/resource_aws_vpc_endpoint_test.go [hunk 2]
```diff
@@ -179,8 +179,7 @@ func TestAccAWSVpcEndpoint_gatewayWithRouteTableAndPolicy(t *testing.T) {
 func TestAccAWSVpcEndpoint_gatewayPolicy(t *testing.T) {
 	var endpoint ec2.VpcEndpoint
 	// This policy checks the DiffSuppressFunc
-	policy1 := `
-{
+	policy1 := `{
```
Comment (bflad) [inline aws/resource_aws_vpc_endpoint_test.go:182]: It appears this policy is being directly injected into heredoc syntax so either the newlines need to be present here or in the test configuration:

```
=== CONT  TestAccAWSVpcEndpoint_gatewayPolicy
TestAccAWSVpcEndpoint_gatewayPolicy: resource_aws_vpc_endpoint_test.go:217: Step 1/3 error: terraform failed: exit status 1
stderr:
Error: Invalid expression
on config241249749/terraform_plugin_test.tf line 15, in resource "aws_vpc_endpoint" "test":
15:   policy       = <<POLICY{
Expected the start of an expression, but found an invalid expression token.
--- FAIL: TestAccAWSVpcEndpoint_gatewayPolicy (1.55s)
```

**Review by bflad — CHANGES_REQUESTED** [review]
Two new test failures, but otherwise looks good. 👍

File: aws/resource_aws_vpc_test.go (hunk 1 above)
Comment (YakDriver) [reply aws/resource_aws_vpc_test.go:579]:
```
--- PASS: TestAccAWSVpc_basic (29.15s)
```

File: aws/resource_aws_vpc_endpoint_test.go (hunk 2 above)
Comment (YakDriver) [reply aws/resource_aws_vpc_endpoint_test.go:182]:
```
--- PASS: TestAccAWSVpcEndpoint_gatewayPolicy (70.16s) (GovCloud)
--- PASS: TestAccAWSVpcEndpoint_gatewayPolicy (61.74s) (commercial)
```

**Review by bflad — APPROVED** [review]

**Pull request merged**